test/mocks/network/dns_service.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
```

Comment:
I would move this next to the test or inside the test. This is not a generic mock that will be used anywhere else.

Reply:
🆗